Sources/XCTest/Public/XCTestCase.swift

```diff
@@ -225,33 +224,61 @@ open class XCTestCase: XCTest {
             }
         }
 
-        setUp()
-    }
-
-    private func performTearDownSequence() {
-        runTeardownBlocks()
-
-        tearDown()
+        do {
+            if #available(macOS 12.0, *) {
+                try awaitUsingExpectation {
+                    try await self.setUp()
+                }
+            }
+        } catch {
+            handleErrorDuringSetUp(error)
+        }
 
         do {
-            try tearDownWithError()
+            try self.setUpWithError()
         } catch {
+            handleErrorDuringSetUp(error)
+        }
+
+        self.setUp()
+    }
+    
+    
+    private func performTearDownSequence() {
+        func handleErrorDuringTearDown(_ error: Error) {
             if error.xct_shouldRecordAsTestFailure {
                 recordFailure(for: error)
             }
         }
-    }
+        
+        func runTeardownBlocks(errorHandler: @escaping (Error) -> Void) {
+            for block in self.teardownBlocksState.finalize().reversed() {
+                do {
+                    try block()
+                } catch {
+                    errorHandler(error)
```

Comment:
I don't think we need the `errorHandler` closure to be marked `@escaping` anymore, since it's no longer being passed to a `Task {}`

Oh and actually! We don't even need it to be a parameter; we can simply reference `func handleErrorDuringTearDown` above directly
```suggestion
        func runTeardownBlocks() {
            for block in self.teardownBlocksState.finalize().reversed() {
                do {
                    try block()
                } catch {
                    handleErrorDuringTearDown(error)
```